### ISSUE TITLE
pg_restore error when sys.replace is used in a computed column

### ISSUE THREAD

**Issue by staticlibs** [wiltondb/wiltondb#59]
Table like this:

```tsql
create database test1
use test1
create table tab1 (
    col1 varchar(8),
    col2 as replace(col1, ' ', '')
)
insert into tab1 (col1) values ('foo bar')
```

cannot be restored with `pg_restore`, fails with:

```
pg_restore: processing data for table "test1_dbo.tab1"
pg_restore: while PROCESSING TOC:
pg_restore: from TOC entry 14499; 0 296013 TABLE DATA tab1 test1_dbo
pg_restore: error: COPY failed for table "tab1": ERROR:  invalid input syntax for type integer: "ig"
LINE 1: ..._replace(input_string, '***=' || pattern, replacement, 'ig')
                                                                  ^
QUERY:  regexp_replace(input_string, '***=' || pattern, replacement, 'ig')
CONTEXT:  PL/pgSQL function sys.replace(text,text,text) line 8 at RETURN
COPY address, line 1: "foo bar"
Error: restore failed, cleaning up global roles we created: test1_db_owner, test1_dbo, test1_guest
```

**Comment by staticlibs** [comment]
This is the [upstream Babelfish issue](https://github.com/babelfish-for-postgresql/babelfish_extensions/issues/2944), the following can be run in `wilton` DB on Postgres connection as a permanent workaround (going to be included into the upcoming release):

```sql
CREATE OR REPLACE FUNCTION sys.replace (in input_string text, in pattern text, in replacement text) returns TEXT as
$body$
begin
   if pattern is null or replacement is null then
       return null;
   elsif pattern = '' then
       return input_string;
   elsif sys.is_collated_ci_as(input_string) then
       return regexp_replace(input_string, '***=' || pattern, replacement, 'ig'::text);
   else
       return regexp_replace(input_string, '***=' || pattern, replacement, 'g'::text);
   end if;
end
$body$
LANGUAGE plpgsql IMMUTABLE PARALLEL SAFE STRICT;

CREATE OR REPLACE FUNCTION sys.replace (input_string sys.VARCHAR, pattern sys.VARCHAR, replacement sys.VARCHAR) returns sys.VARCHAR as
$body$
begin
   if pattern is null or replacement is null then
       return null;
   elsif pattern = '' then
       return input_string;
   elsif sys.is_collated_ci_as(input_string) then
       return regexp_replace(input_string, '***=' || pattern, replacement, 'ig'::text);
   else
       return regexp_replace(input_string, '***=' || pattern, replacement, 'g'::text);
   end if;
end
$body$
LANGUAGE plpgsql IMMUTABLE PARALLEL SAFE STRICT;

CREATE OR REPLACE FUNCTION sys.replace (input_string sys.NVARCHAR, pattern sys.NVARCHAR, replacement sys.NVARCHAR) returns sys.NVARCHAR as
$body$
begin
   if pattern is null or replacement is null then
       return null;
   elsif pattern = '' then
       return input_string;
   elsif sys.is_collated_ci_as(input_string) then
       return regexp_replace(input_string, '***=' || pattern, replacement, 'ig'::text);
   else
       return regexp_replace(input_string, '***=' || pattern, replacement, 'g'::text);
   end if;
end
$body$
LANGUAGE plpgsql IMMUTABLE PARALLEL SAFE STRICT;
```

**Comment by staticlibs** [comment]
The fix is included in [update 12.16.1](https://github.com/wiltondb/wiltondb/releases/tag/3-lts-12-16-1), closing as completed.
